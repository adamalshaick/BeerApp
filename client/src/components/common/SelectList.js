import React from "react";
import PropTypes from "prop-types";

const SelectList = ({ name, value, onChange, options }) => {
  const selectOptions = options.map(option => (
    <option key={option.label} value={option.value}>
      {option.label}
    </option>
  ));

  return (
    <select
      className="form-control"
      name={name}
      value={value}
      onChange={onChange}
    >
      {selectOptions}
    </select>
  );
};

SelectList.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  options: PropTypes.array.isRequired
};

export default SelectList;
