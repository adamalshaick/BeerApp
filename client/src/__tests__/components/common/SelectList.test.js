import React from "react";
import SelectList from "../../../components/common/SelectList";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

const Props = {
  options: [{ label: "label", value: "value" }]
};

const wrapper = shallow(<SelectList {...Props} />);
describe("render component", () => {
  it("renders component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
