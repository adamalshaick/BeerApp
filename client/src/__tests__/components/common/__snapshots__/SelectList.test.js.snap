// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
<select
  className="form-control"
>
  <option
    key="label"
    value="value"
  >
    label
  </option>
</select>
`;
